clippy_lints/src/format_args.rs

```diff
@@ -86,9 +153,69 @@ impl<'tcx> LateLintPass<'tcx> for FormatArgs {
                     check_format_in_format_args(cx, outermost_expn_data.call_site, name, arg.param.value);
                     check_to_string_in_format_args(cx, name, arg.param.value);
                 }
+                if meets_msrv(self.msrv, msrvs::FORMAT_ARGS_CAPTURE) {
+                    check_uninlined_args(cx, &format_args, outermost_expn_data.call_site);
+                }
             }
         }
     }
+
+    extract_msrv_attr!(LateContext);
+}
+
+fn check_uninlined_args(cx: &LateContext<'_>, args: &FormatArgsExpn<'_>, call_site: Span) {
```

Comment:
You can add a check for `args.format_string.span.from_expansion()`, that'll catch the case where the string comes from a macro (excluding the weird proc macro case)

Reply:
@Alexendoo thanks, it does fix the `println!(no_param_str!(), local_i32);` case, where the macro expands into a format string, but it doesn't handle these cases with the pass_through macro (`from_expansion` returns false):

```rust
    macro_rules! pass_through { ($expr:expr) => { $expr }; }
    ...
    println!(pass_through!("foo={local_i32}"), local_i32 = local_i32);
    println!(pass_through!("foo={}"), local_i32);
```

Reply:
Yeah in that example there's no distinguishing the span of `pass_through!("str")` and `"str"`

Reply:
@alexendoo ok, thanks! How would you like to proceed? We could either merge it as a pedantic lint due to false positives, and fix it later, or postpone it and possibly make it as style once fixed?

Reply:
My plan is to merge it, I've kept it waiting long enough 😄